src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1385,20 +1385,23 @@ private LocalDefinition GetLocal(LocalSymbol symbol)
 
         private LocalDefinition DefineLocal(LocalSymbol local, SyntaxNode syntaxNode)
         {
-            var transformFlags = default(ImmutableArray<TypedConstant>);
-            bool hasDynamic = local.Type.ContainsDynamic();
-            var isDynamicSourceLocal = hasDynamic && !local.IsCompilerGenerated;
-            if (isDynamicSourceLocal)
-            {
-                NamedTypeSymbol booleanType = _module.Compilation.GetSpecialType(SpecialType.System_Boolean);
-                transformFlags = CSharpCompilation.DynamicTransformsEncoder.Encode(local.Type, booleanType, 0, RefKind.None);
-            }
+            var dynamicTransformFlags = !local.IsCompilerGenerated && local.Type.ContainsDynamic() ?
+                CSharpCompilation.DynamicTransformsEncoder.Encode(local.Type, _module.Compilation.GetSpecialType(SpecialType.System_Boolean), 0, RefKind.None) :
+                ImmutableArray<TypedConstant>.Empty;
+            var tupleElementNames = !local.IsCompilerGenerated && local.Type.ContainsTuple() ?
```

Comment:
Yes, thanks. Changed here and in `EEMethodSymbol.AddSynthesizedReturnTypeAttributes`.